src/components/map/component.js

```diff
@@ -145,6 +153,20 @@ class Map extends Component {
     });
   };
 
+  onHover = () => {
+    this.setState({ showPopup: true });
+    setTimeout(() => {
```

Comment:
I'm not sure this is the best way of doing it... try to understand it. You are setting the state on every single moment you hover the map. Apart from that, every time you set the state you run a timeout that will set the state again. This is really wrong